### DISCUSSION TITLE
Problem about input in Raspberry OS Lite with touch screen.

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#12031] [Q&A] [answered]
I followed this tutorial(https://docs.avaloniaui.net/docs/next/guides/platforms/rpi/running-on-raspbian-lite-via-drm) to deploy in Raspberry OS Lite and it runs well.

Now I met a new problem that I cannot input anything in TextBox control with touch screen.

I search and get to know there is some virtual keyboard in Linux, just like https://raspberrytips.com/install-virtual-keyboard-raspberry-pi/

However, it seems all of them can only run under desktop of Linux. In spite I installed it as the tutorial said, it doesn't work any.

Then, I found this discussion: #https://github.com/AvaloniaUI/Avalonia/discussions/11678

I am not sure but it seems it cannot use a third-party virtual keyboard in Avalonia but can only achieve this by embedding a virtual keyboard in program.

I am not quite sure about this. What's more, I need to input Chinese that I have to not only find a virtual keyboard but also an IME to achieve this. I have absolutely no idea how to solve this problem.  Can anyone give me a suggestion? Thank you.

**Comment by AliAbed93** [comment, marked as answer]
if you can run Avalonia on Raspberry pi headless OS, then you can write your touch keyboard for yourself by XAML!
